docs/developer/releasing.md

```diff
@@ -0,0 +1,42 @@
+# Releasing
+
+## Output
+
+### Expected artifacts
+
+1. A container image of the shared cluster-api controller manager
+2. A git tag for providers to use
+
+### Artifact locations
+
+1. The container image is found in the registry `gcr.io/k8s-cluster-api` with an
+   image name of `cluster-api-controller` and a tag that matches the release
+   version. For example, in the `0.0.0-alpha.4` release, the container image
+   location is `gcr.io/k8s-cluster-api/cluster-api-controller:0.0.0-alpha.4`
+
+## Process
+
+1. Create a pull request that contains two changes:
+   1. Change [the cluster api controller manager image
+   tag][managerimg] from `:latest` to whatever version is being released
+   2. Change the `CONTROLLER_IMAGE` variable in the [Makefile][makefile] to the
+      version being released
+2. Get the pull request merged
+3. From the commit in step 1 (that is now in the master branch), build and push
+   the container image with `make docker-push`
+4. Create a tag from this same commit and push the tag to the github repository
+5. Revert the commit made in step 1
+6. Open a pull request with the revert change
+7. Get that pull request merged
+8. Create a release in github based on the tag created above
+
```

Comment:
Step 9 is to write release notes. Right now this is also a very manual process. For https://github.com/kubernetes-sigs/cluster-api/releases/tag/0.0.0-alpha.3 I basically went through all of the PRs that had merged since the prior release and manually curated the release notes. It's onerous, but the script in the k/k repo that does it didn't seem like it would work out of the box even though we use the same release notes block in our PRs. As you can tell, when I cut https://github.com/kubernetes-sigs/cluster-api/releases/tag/0.0.0-alpha.4 during kubecon I didn't have the time to make similarly thorough release notes. 

I've filed https://github.com/kubernetes-sigs/cluster-api/issues/327 to track building better automation for release note generation.